src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -631,61 +521,37 @@ bool metadataTypeMatches(SymbolSet s, INamedTypeSymbol t)
                 // our criteria.
```

Comment:
great question!  first, know that this is an index over metadata, and that index is built independent of that DLL's context in some containing project.  THe way the index is built is to actually crack it open directly, read the tables and build information out of that.  So the index doesn't store "this ISymbol derives from this other ISymbol", instead it stores "I remember that a type called 'FileStream' derived from a type called 'Stream'".  

The goal of hte index was to be fast and lightweight and unencumbered from concerns like retargetting.  Instead, it just stores basic data pulled out of the metadata tables in a lightweight fashion.  

It's actually just like how the source-index works, just a little faster and more accurate because we can look at real md tokens.